### PR TITLE
Create The API for checking the GCS config

### PR DESCRIPTION
**Pull request by daveaugustus**
### :nut_and_bolt: Description: What code changed, and why?
Create The API for checking the GCS config

### :chains: Related Resources
https://chefio.atlassian.net/jira/software/c/projects/CHEF/boards/482?selectedIssue=CHEF-5613
### :athletic_shoe: How to Build and Test the Change
- Run verify server
- Trigger the following command with req body
```
curl --location 'http://127.0.0.1:7799/api/v1/checks/gcp-cloud-storage-config' \
--header 'Content-Type: application/json' \
--data '{
   
}'
```
### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Screenshot 2023-08-31 at 2 07 43 AM

### CODE REVIEW

File: components/automate-cli/pkg/verifyserver/utils/gcputils/gcp.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	"fmt"
+
+	"cloud.google.com/go/storage"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+type GCPUtils interface {
```
Comment (Dmaddu) [inline components/automate-cli/pkg/verifyserver/utils/gcputils/gcp.go:15]: Please make the changes as discussed

**Review by Dmaddu — CHANGES_REQUESTED** [review]

File: components/automate-cli/pkg/verifyserver/services/gcpcloudstorageservice/gcpconfigservice.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+func NewGCPCloudStorageConfig(logger logger.Logger, gcpUtils gcputils.GCPUtils) GCPCloudStorageConfig {
+	return &GCPConfigService{
+		Logger:   logger,
+		GCPUtils: gcpUtils,
+	}
+}
+
+func (ss *GCPConfigService) GetGCPConnection(req *models.GCPCloudStorageConfigRequest) *models.Checks {
+	ss.Req = req
+	ctx := context.Background()
```
Comment (YashviJain01) [inline components/automate-cli/pkg/verifyserver/services/gcpcloudstorageservice/gcpconfigservice.go:38]: Should we get this context from the Call itself?

File: cspell.json [hunk 3]
```diff
@@ -142,7 +142,6 @@
         "dups",
         "eaaccess",
         "EAS",
-        "opensearch",
```
Comment (YashviJain01) [inline cspell.json:145]: Why is this file here? Do we need this?

File: components/automate-cli/pkg/verifyserver/constants/gcpconfig.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package constants
+
+const (
+	GCP_CONNECTION_TITLE                = "GCP connection test"
+	GCP_BUCKET_NOT_FOUND                = "Cannot find the Bucket in GCP cloud storage"
```
Comment (YashviJain01) [inline components/automate-cli/pkg/verifyserver/constants/gcpconfig.go:5]: Should we rename GCP Cloud Storage to GCS?

File: components/automate-cli/pkg/verifyserver/services/gcpcloudstorageservice/gcpconfigservice.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/gcputils"
+	"github.com/chef/automate/lib/logger"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var filePrefix = "gcp_check_test_"
```
Comment (YashviJain01) [inline components/automate-cli/pkg/verifyserver/services/gcpcloudstorageservice/gcpconfigservice.go:16]: Can we move this to constants file?

**Comment by YashviJain01** [comment]
Can you check the pipelines as well?

File: components/automate-cli/pkg/verifyserver/utils/gcputils/gcp.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+func (au *GCPUtilsImpl) NewSessionWithOptions(ctx context.Context, gsa *models.GcpServiceAccount) (*storage.Client, error) {
+	bx, _ := json.Marshal(gsa)
+	return storage.NewClient(ctx, option.WithCredentialsJSON(bx))
+}
+
+func (au *GCPUtilsImpl) NewUploader(ctx context.Context, uploadObject *storage.ObjectHandle) (*models.Checks, error) {
+	wc := uploadObject.NewWriter(ctx)
+	_, err := fmt.Fprintf(wc, "Heute ist ein schöner Tag.")
```
[the remark this answers is not in the document]
Comment (daveaugustus) [reply components/automate-cli/pkg/verifyserver/utils/gcputils/gcp.go:37]: "Today is a beautiful day"

**Comment by daveaugustus** [comment]
https://github.com/chef/automate/assets/36985548/686e8eed-dc61-4df6-a0ba-3df1aa58d497

@Dmaddu please find the testing video

**Pull request merged**